File: server/models/businessClosure.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require("../config/db");

const BusinessClosure = sequelize.define(
    "business_closures",
    {
        id: {
            type: DataTypes.BIGINT,
            primaryKey: true,
            autoIncrement: true,
        },
        business_id: {
            type: DataTypes.BIGINT,
            allowNull: false
        },
        title: {
            type: DataTypes.STRING,
            allowNull: false
        },
        start_date: {
            type: DataTypes.DATEONLY,
            allowNull: false
        },
        end_date: {
            type: DataTypes.DATEONLY,
            allowNull: false
        },
        start_time: {
            type: DataTypes.TIME,
            allowNull: true
        },
        end_time: {
            type: DataTypes.TIME,
            allowNull: true
        },
        is_all_day: {
            type: DataTypes.BOOLEAN,
            defaultValue: true
        },
        reason: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        status: {
            type: DataTypes.BOOLEAN,
            defaultValue: true
        },
        created_at: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        },
        updated_at: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        }
    },
    {
        timestamps: false,
        tableName: "business_closures",
        indexes: [
            { fields: ["business_id"] },
            { fields: ["start_date", "end_date"] },
            { fields: ["status"] }
        ]
    }
);

module.exports = BusinessClosure;

File: my_booking_application/lib/app/data/models/business_closure_model.dart
class BusinessClosureModel {
  int? id;
  int? businessId;
  String? title;
  String? startDate;
  String? endDate;
  String? startTime;
  String? endTime;
  bool? isAllDay;
  String? reason;

  BusinessClosureModel({
    this.id,
    this.businessId,
    this.title,
    this.startDate,
    this.endDate,
    this.startTime,
    this.endTime,
    this.isAllDay,
    this.reason,
  });

  BusinessClosureModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    businessId = json['business_id'];
    title = json['title'];
    startDate = json['start_date'];
    endDate = json['end_date'];
    startTime = json['start_time'];
    endTime = json['end_time'];
    isAllDay = json['is_all_day'] == true || json['is_all_day'] == 1 || json['is_all_day'] == '1';
    reason = json['reason'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'business_id': businessId,
      'title': title,
      'start_date': startDate,
      'end_date': endDate,
      'start_time': startTime,
      'end_time': endTime,
      'is_all_day': isAllDay,
      'reason': reason,
    };
  }
}
